[Bug Report] wxPython Thread error

This part of the code throws an error

https://github.com/Amulet-Team/Amulet-Map-Editor/blob/0.10/amulet_map_editor/programs/edit/api/canvas/base_edit_canvas.py#L162-L169

```
Traceback (most recent call last):
  File "amulet_map_editor\programs\edit\api\canvas\base_edit_canvas.py", line 148, in thread_setup
  File "minecraft_model_reader\api\resource_pack\java\download_resources.py", line 99, in get_java_vanilla_latest_iter
  File "minecraft_model_reader\api\resource_pack\java\download_resources.py", line 63, in get_latest_iter
  File "minecraft_model_reader\api\resource_pack\java\download_resources.py", line 56, in get_latest_iter
  File "minecraft_model_reader\api\resource_pack\java\download_resources.py", line 27, in get_launcher_manifest
  File "json\__init__.py", line 293, in load
  File "http\client.py", line 474, in read
  File "http\client.py", line 624, in _safe_read
  File "socket.py", line 589, in readinto
  File "ssl.py", line 1071, in recv_into
  File "ssl.py", line 929, in read
socket.timeout: The read operation timed out

Traceback (most recent call last):
  File "amulet_map_editor\programs\edit\api\canvas\base_edit_canvas.py", line 148, in thread_setup
  File "minecraft_model_reader\api\resource_pack\java\download_resources.py", line 99, in get_java_vanilla_latest_iter
  File "minecraft_model_reader\api\resource_pack\java\download_resources.py", line 63, in get_latest_iter
  File "minecraft_model_reader\api\resource_pack\java\download_resources.py", line 56, in get_latest_iter
  File "minecraft_model_reader\api\resource_pack\java\download_resources.py", line 27, in get_launcher_manifest
  File "json\__init__.py", line 293, in load
  File "http\client.py", line 474, in read
  File "http\client.py", line 624, in _safe_read
  File "socket.py", line 589, in readinto
  File "ssl.py", line 1071, in recv_into
  File "ssl.py", line 929, in read
socket.timeout: The read operation timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "amulet_map_editor\programs\edit\edit.py", line 105, in _thread_setup
  File "amulet_map_editor\programs\edit\edit.py", line 75, in _update_loading
  File "amulet_map_editor\programs\edit\api\canvas\base_edit_canvas.py", line 168, in thread_setup
wx._core.wxAssertionError: C++ assertion "wxThread::IsMain()" failed at ..\..\src\msw\evtloop.cpp(176) in wxGUIEventLoop::Dispatch(): only the main thread can process Windows messages
```

--- a/amulet_map_editor/programs/edit/api/canvas/base_edit_canvas.py
+++ b/amulet_map_editor/programs/edit/api/canvas/base_edit_canvas.py
@@ -8,6 +8,7 @@ import os
 from typing import Optional, Generator
 import weakref
 import sys
+import time
 
 from minecraft_model_reader.api.resource_pack.java.download_resources import (
     get_java_vanilla_latest_iter,
@@ -159,14 +160,25 @@ class BaseEditCanvas(EventCanvas):
                     msg,
                     exc_info=True,
                 )
-                dialog = TracebackDialog(
-                    self,
-                    lang.get("program_3d_edit.canvas.java_rp_failed"),
-                    f"{msg}\n{e}",
-                    traceback.format_exc(),
-                )
-                dialog.ShowModal()
-                dialog.Destroy()
+                wait = True
+
+                def show_error():
+                    nonlocal wait
+                    try:
+                        dialog = TracebackDialog(
+                            self,
+                            lang.get("program_3d_edit.canvas.java_rp_failed"),
+                            f"{msg}\n{e}",
+                            traceback.format_exc(),
+                        )
+                        dialog.ShowModal()
+                        dialog.Destroy()
+                    finally:
+                        wait = False
+
+                wx.CallAfter(show_error)
+                while wait:
+                    time.sleep(0.1)
 
             yield 0.5, lang.get("program_3d_edit.canvas.loading_resource_packs")
             packs += [pack for pack in user_packs if isinstance(pack, JavaResourcePack)]

